app/models/user.rb

```diff
@@ -75,7 +75,7 @@ def student
   accepts_nested_attributes_for :roles, allow_destroy: true
 
   before_save :normalize_location
-  after_create :complete_from_github
+  before_create :complete_from_github
```

Comment:
This is more efficient, as it will only do the save once. (I couldn't think of any reason why you'd want to do it twice.)

Reply:
I fear I don't understand the reasoning here 🙈 
Do you mean it's more efficient, since under the hood `complete_from_github` will persist a new version of the record?

Reply:
@klappradla I'm worried about this change. I didn't had the chance to dive into it today. But I wouldn't be surprised if this was intended to be `after_create` because of some other delicacies. And I'm not sure that it's covered by tests. 
I find that more worrying than the ineffeciency.

Reply:
I'm pretty confident on this one, but you folks know this codebase much better than I do.

The logic in the hook was just to add some additional details to the model, and fetched from GitHub. Any GitHub errors are already handled, so the method should never fail. I suppose that validations *could* fail on the updated attributes, and having this as `after_create` rather than `before_create` would yield different behaviour in that case - is that something you might care about here?

What I'm not sure of is why the tests were failing on Rails 5.2.1 without the change. For some reason the `email` wasn't getting set when this was an `after_create` but it is now that it's switched. That was the main motivation for me changing it.

Reply:
I already made a note to myself to use my next holidays to actually make it covered by tests @F3PiX 😉 

I get your concerns and actually had the same. But I could not find an obvious reason why doing it this way should not work. The flow does not include any other `after_commit` or anything hooks, so it should all even work without persisting 🤔 
Looking at it, I would probably even have hooked in before validating - since we're actually changing some attributes.

Reply:
I'm still worried. It's not the obvious things I am worried about, but hidden side effects.:
> For some reason the email wasn't getting set....  
It's the 'for some reason' that worries me. Why was this affected by the upgrade to 5.2? 
IF this is going to have side effects that we're not aware of, it will not be manifest until next season, when new users are created and it will be a pain to trace it back to this change.

Reply:
Makes sense @F3PiX ! So let's roll back this PR and look into the details _"by hand"_ 👀 

I fear I won't have time for the "looking into" part within the next weeks though. I still roll back this PR. If you're fine I can do it later today - better to not have too many other changes depend on it.

Reply:
👍 
I wished I had time to look into it, but same here: I won't have time before the end of the year :-(
Thanks for rolling back, now I stop worrying :-)